### PR TITLE
[2015.2] Push extra kwargs deprecation to Carbon

### PR DESCRIPTION
**Pull request by s0undt3ch**

### CODE REVIEW

File: salt/utils/__init__.py [hunk 1]
```diff
@@ -826,6 +826,17 @@ def format_call(fun,
             continue
         extra[key] = copy.deepcopy(value)
 
+    # We'll be showing errors to the users until Salt Lithium comes out, after
```
Comment (jfindlay) [inline salt/utils/__init__.py]: did you mean 'Salt Beryllium' here?

Comment (s0undt3ch) [inline salt/utils/__init__.py]: Didn't replace that one.

**Comment by s0undt3ch** [comment]
This PR is not yet complete. Need to fix the outputters.

I'll let you know when...

**Comment by s0undt3ch** [comment]
This is a blocker for Beryllium

**Comment by s0undt3ch** [comment]
Ready for review once the tests pass

**Comment by s0undt3ch** [comment]
Correction, this is a blocker for Lithium.

**Comment by s0undt3ch** [comment]
Lint failure is unrelated.

**Pull request merged**